http/http_test.go

```diff
@@ -14,3 +17,40 @@ func TestIsLocalAddress(t *testing.T) {
 	assert.True(t, isLocalAddress("172.16.0.0"))
 	assert.True(t, isLocalAddress("169.254.169.254"))
 }
+
+type countServer struct {
+	count int
+}
+
+func (c *countServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	c.count++
+	w.Write([]byte("Done"))
+}
+
+func TestSafeHTTPClient(t *testing.T) {
+	client := &http.Client{}
+
+	counter := &countServer{}
```

Comment:
I'd use the `httptest` lib instead. https://golang.org/pkg/net/http/httptest/#Server